spec/integration/parser/epp_parameters_spec.rb

```diff
@@ -0,0 +1,231 @@
+require 'spec_helper'
+require 'puppet_spec/language'
+
+module PuppetSpec
+module EppRunner
+  extend RSpec::Matchers::DSL
+
+  def self.collect_log(source, args, code)
+    Puppet[:code] = code
+    compiler = Puppet::Parser::Compiler.new(Puppet::Node.new('specification'))
+    logs = []
+    Puppet::Util::Log.with_destination(Puppet::Test::LogCollector.new(logs)) do
+      compiler.compile
+      Puppet::Pops::Evaluator::EppEvaluator.inline_epp(compiler.topscope, source, args)
+    end
+    logs = logs.select { |log| log.level == :notice }.map { |log| log.message }
+    logs
+  end
+
+  def run_expectations(decl, expectations, code = 'undef')
+    expectations.each do |args, result|
+      if result[0].is_a?(Class)
+        it "'#{decl}' called with #{args} should fail with error #{result}" do
+          expect {EppRunner.collect_log(decl, args, code)}.to raise_error(result[0], result[1])
+        end
+      else
+        it "'#{decl}' called with #{args} should produce #{result}" do
+          expect(EppRunner.collect_log(decl, args, code)).to include(*result)
+        end
+      end
+    end
+  end
+end
+
+describe "EPP parameter default expressions" do
+  extend EppRunner
+
+  context "to the left of a parameter can use that parameter's value" do
+    run_expectations(<<-SOURCE.gsub(/^ {6}/, ''),
+      <%|
+        $a = 10,
+        $b = $a
+      |%>
+      <%
+        notice($a)
+        notice($b)
+      %>
+      SOURCE
+      [ [{}, ['10', '10']],
```

Comment:
I would like this to read:
```
[ [ "when no args are given, param lookup finds default value",  {}, ['10', '10']],
  ["when one arg is given, param lookup finds given value", {'a' => 2}, ['2', '2']],
  ["when two args are given, given value is used for both", {'a' => 2, 'b' => 5}, ['2', '5'] ]])
```

So that those strings can be included in the "it" part in the generic method